spring-kafka/README.md

```diff
@@ -0,0 +1,357 @@
+        <entry key="org.springframework.kafka.core.DefaultKafkaProducerFactory" value="org.springframework.kafka.core.ProducerFactory" />
+
+        <entry key="org.springframework.kafka.core.KafkaTemplate" value="org.springframework.kafka.core.KafkaTemplate" />
+      </map>
+    </argument>
+  </bean>
+</type-converters>
+
+<!-- Producer properties -->
+<bean id="kafkaProducerProperties" class="java.util.HashMap">
+  <argument>
+    <map key-type="java.lang.String" value-type="java.lang.Object">
+      <entry key="bootstrap.servers" value="localhost:9092"/>
+      <entry key="retries" value="0"/>
+      <entry key="batch.size" value="16384"/>
+      <entry key="linger.ms" value="1"/>
+      <entry key="buffer.memory" value="33554432"/>
+    </map>
+  </argument>
+</bean>
+
+<!-- Producer Kafka Template -->
+<bean id="kafkaProxyProducerKafkaTemplate" class="org.springframework.kafka.core.KafkaTemplate">
+  <argument> <!-- type converter used on this argument -->
+    <bean class="org.springframework.kafka.core.DefaultKafkaProducerFactory">
+      <argument ref="kafkaProducerProperties"/>
+      <argument> <!-- type converter used on this argument -->
+        <bean class="org.apache.kafka.common.serialization.StringSerializer" />
+      </argument>
+      <argument> <!-- type converter used on this argument -->
+        <bean class="org.hitachivantara.spring.kafka.protobuf.KafkaProtobufSerializer" />
+      </argument>
+    </bean>
+  </argument>
+  <argument type="java.lang.Boolean" value="true" /> <!-- auto flush -->
+</bean> 
+```
+
+A `BaseMessageProducer` class was added to be able to produce messages just by defining a bean for that class. That way the usage of 
+the Kafka Template in our code is black boxed. To be able to use it a bean like this must be defined:
+```xml
+<bean id="mySimpleKafkaProducerBean" class="org.hitachivantara.spring.kafka.producer.BaseMessageProducer">
+  <property name="kafkaTemplate" ref="kafkaProxyProducerKafkaTemplate"/>
+  <property name="kafkaTopic" value="myKafkaTopic"/>
+</bean>
+```
+
+### Type converter
+Notice that the `org.springframework.kafka.core.KafkaTemplate` entry in the type converter is needed in order to 
+inject the example bean `kafkaProxyProducerKafkaTemplate` in our applications beans like this:
+```xml
+  <bean id="ourApplicationBean" class="com.hitachivantara.ApplicationBean">
+    <property name="kafkaTemplate" ref="kafkaProxyProducerKafkaTemplate"/> <!-- type converter used on this argument -->
+  </bean>
+``` 
+```java
+public class ApplicationBean {
+  private KafkaTemplate<String, Message> kafkaTemplate;
+  
+  public void setKafkaTemplate( KafkaTemplate<String, Message> kafkaTemplate ) {
+    this.kafkaTemplate = kafkaTemplate;
+  }
+}
+```
+
+### KafkaTemplate
+The easiest way to send messages is using the Spring Kafka Template API defined in `org.springframework.kafka.core.KafkaTemplate`.
```

Comment:
Explain overall what is the `KafkaTemplate`, meaning what is its role, and provide a link to the spring documentation. It might eventually make sense to talk about it in the big picture introduction.